action_test.go

```diff
@@ -0,0 +1,200 @@
+package hbase
+
+import (
+	pb "github.com/golang/protobuf/proto"
+	. "github.com/pingcap/check"
+	"github.com/pingcap/go-hbase/proto"
+)
+
+type mockAction struct {
+}
+
+func (m *mockAction) ToProto() pb.Message {
+	return &mockMessage{}
+}
+
+type mockMessage struct {
+}
+
+func (m *mockMessage) Reset() {
+}
+
+func (m *mockMessage) String() string {
+	return "mock message"
+}
+
+func (m *mockMessage) ProtoMessage() {
+}
+
+type Message interface {
+	Reset()
+	String() string
+	ProtoMessage()
+}
+
+type ActionTestSuit struct {
+	cli       HBaseClient
+	tableName string
+}
+
+var _ = Suite(&ActionTestSuit{})
+
+func (s *ActionTestSuit) SetUpTest(c *C) {
+	var err error
+	s.cli, err = NewClient(getTestZkHosts(), "/hbase")
+	c.Assert(err, IsNil)
+
+	s.tableName = "test_action"
+	tblDesc := NewTableDesciptor(NewTableNameWithDefaultNS(s.tableName))
+	cf := NewColumnFamilyDescriptor("cf")
+	tblDesc.AddColumnDesc(cf)
+	s.cli.CreateTable(tblDesc, nil)
+}
+
+func (s *ActionTestSuit) TearDownTest(c *C) {
+	err := s.cli.DisableTable(NewTableNameWithDefaultNS(s.tableName))
+	c.Assert(err, IsNil)
+
+	err = s.cli.DropTable(NewTableNameWithDefaultNS(s.tableName))
+	c.Assert(err, IsNil)
+}
+
+func (s *ActionTestSuit) TestAction(c *C) {
```

Comment:
Could you split this test case into serveral test cases.
Someday if it failed we still can't know how to debug it.